lightning/src/chain/mod.rs

```diff
@@ -57,6 +57,81 @@ pub trait Listen {
+///
+/// Clients should use this interface instead of [`Listen`] if they receive transaction data only
+/// relevant to that identified by [`Filter`]. For instance, an Electrum client may implement
+/// [`Filter`] by subscribing to activity related to registered transactions and outputs. Upon
+/// notification, it would pass along the relevant transactions using this interface.
+///
+/// The intended use is as follows:
+/// - Call [`transactions_confirmed`] for any on-chain activity of interest.
+/// - Return such transactions from [`get_relevant_txids`] until they have enough confirmations to
+///   no longer be in danger of chain reorganization.
+/// - Call [`transaction_unconfirmed`] for any transaction returned by [`get_relevant_txids`] that
+///   has been reorganized out of the chain.
+/// - Call [`best_block_updated`] whenever a new chain tip becomes available.
+///
+/// See individual method documentation for further details.
+///
+/// [`transactions_confirmed`]: Self::transactions_confirmed
+/// [`transaction_unconfirmed`]: Self::transaction_unconfirmed
+/// [`best_block_updated`]: Self::best_block_updated
+/// [`get_relevant_txids`]: Self::get_relevant_txids
+pub trait Confirm {
+	/// Processes transactions confirmed in a block with a given header and height.
+	///
+	/// Should be called for any transactions registered by [`Filter::register_tx`] or any
+	/// transactions spending an output registered by [`Filter::register_output`]. Such transactions
+	/// appearing in the same block do not need to be included in the same call; instead, multiple
+	/// calls with additional transactions may be made.
+	///
+	/// May be called before or after [`best_block_updated`] for the corresponding block. However,
+	/// it must not be called after [`best_block_updated`] if `header` corresponds to a block that
+	/// has been reorganized from the chain (i.e., if the transaction is no longer confirmed).
+	///
+	/// [`best_block_updated`]: Self::best_block_updated
+	fn transactions_confirmed(&self, header: &BlockHeader, txdata: &TransactionData, height: u32);
+
+	/// Processes a transaction that is no longer confirmed as result of a chain reorganization.
+	///
+	/// Should be called for any transaction returned by [`get_relevant_txids`] if it has been
+	/// reorganized out of the best chain. Once called, the given transaction should not be returned
+	/// by [`get_relevant_txids`] unless it has been reconfirmed via [`transactions_confirmed`].
+	///
+	/// [`get_relevant_txids`]: Self::get_relevant_txids
+	/// [`transactions_confirmed`]: Self::transactions_confirmed
+	fn transaction_unconfirmed(&self, txid: &Txid);
+
+	/// Processes an update to the best header connected at the given height.
+	///
+	/// Should be called when a new header is available but may be skipped for intermediary blocks
+	/// if they become available at the same time.
+	///
+	/// Implementations are not required to handle chain reorganizations unless `height` is not
```

Comment:
Hmm, I'm not really sure what the point of this comment is - I can't really figure out what this should mean to me as a user who wishes to call this method on an implementation of this trait?

Reply:
Right, I guess its a question of if we want to support both modes - I *think* right now you can call either `transaction_unconfirmed` or you can skip it as long as you call `best_block_updated` with reorg info. I don't think it really makes sense for clients to do that (if they're fetching the full header chain they should be using `Listen`, eventually), but technically we could support it. If we don't want to support it, I dunno if its worth mentioning in the docs here, but if you have any other reasoning based on your electrum exploration we can.

Reply:
Dropped the paragraph.